adapters/applogy/applogy.go

```diff
@@ -0,0 +1,157 @@
+package applogy
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type ApplogyAdapter struct {
+	endpoint string
+}
+
+func (a *ApplogyAdapter) MakeRequests(request *openrtb.BidRequest, _ *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	impressions := request.Imp
+	totalImpressions := len(impressions)
+	result := make([]*adapters.RequestData, 0, totalImpressions)
+	errs := make([]error, 0, totalImpressions)
+
+	for i, impression := range impressions {
+		if impression.Banner == nil && impression.Video == nil && impression.Native == nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: "Applogy only supports banner, video or native ads",
+			})
+			continue
+		}
+		if impression.Banner != nil {
+			banner := impression.Banner
+			if banner.W == nil || banner.H == nil || *banner.W == 0 || *banner.H == 0 {
+				if len(banner.Format) == 0 {
+					errs = append(errs, &errortypes.BadInput{
+						Message: "banner size information missing",
+					})
+					continue
+				}
+				format := banner.Format[0]
+				banner.W = &format.W
+				banner.H = &format.H
+			}
+		}
+		if len(impression.Ext) == 0 {
+			errs = append(errs, errors.New("impression extensions required"))
+			continue
+		}
+		var bidderExt adapters.ExtImpBidder
+		err := json.Unmarshal(impression.Ext, &bidderExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if len(bidderExt.Bidder) == 0 {
+			errs = append(errs, errors.New("Applogy token required"))
+			continue
+		}
+		var impressionExt openrtb_ext.ExtImpApplogy
+		err = json.Unmarshal(bidderExt.Bidder, &impressionExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if impressionExt.Token == "" {
+			errs = append(errs, errors.New("Applogy token required"))
+			continue
+		}
+		request.Imp = impressions[i : i+1]
+		body, err := json.Marshal(request)
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+		result = append(result, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     a.endpoint + "/" + impressionExt.Token,
+			Body:    body,
+			Headers: headers,
+		})
+	}
```

Comment:
The current logic allows `MakeRequests` to return an empty `result` array in the following two corner scenarios:
   1) `request.Imp` was empty to begin with
   2) `request.Imp` was not empty but none of its elements passed the checks inside the `for` loop and we are returning an empty `result` array and an `errs` array of lenght `len(impressions)`

Do you want to append one more error to `errs` describing this situation and setting `result` to `nil`?